enhancements/authentication/external-oidc-additional-identity-information-sources.md

```diff
@@ -0,0 +1,804 @@
+oc login --exec-plugin="oc-oidc" --client-secret=... --client-id=...
+```
+
+3. End users identify their group memberships:
+
+```sh
+oc apply -o yaml -f - <<EOF
+apiVersion: authentication.k8s.io/v1
+kind: SelfSubjectReview
+metadata:
+  name: test
+EOF
+```
+
+Output:
+```sh
+apiVersion: authentication.k8s.io/v1
+kind: SelfSubjectReview
+metadata:
+  creationTimestamp: "2025-12-09T17:03:01Z"
+status:
+  userInfo:
+    ...
+    groups:
+    - system:authenticated
+    - foo # fetched from UserInfo endpoint
+```
+
+**Scenario**: Sourcing group information from Entra ID that was not provided in the JWT via the Microsoft Graph API.
+
+1. Cluster Administrator configures the `authentications.config.openshift.io/cluster` resource
+to tell OpenShift how to fetch the group information from the identity provider like so:
+```yaml
+apiVersion: config.openshift.io/v1
+kind: Authentication
+metadata:
+  name: cluster
+spec:
+  type: OIDC
+  oidcProviders:
+  - name: foo
+    claimMappings:
+      uid:
+        claim: email
+      username:
+        claim: username
+      groups:
+        claim: groups
+    externalClaims: # This is the proposed new field
+      clientAuth:
+        type: ClientCredential # Use the token from the request as the access token to the UserInfo endpoint
+        clientCredential:
+          id: "kas"
+          secret: "kas-foo-client-secret"
+      claims:
+      - method: GET
+        url:
+          type: Expression
+          expression: "\"https://graph.microsoft.com/v1.0/users/\" + claims.upn + \"/memberOf\""
+        mappings:
+          - name: groups
+            expression: "has(response.value) ? response.value.map(x, x.displayName).join(',') : \"\""
+```
+2. End user authenticates with Entra ID and gets an access token:
+
+```sh
+oc login --exec-plugin="oc-oidc" --client-secret=... --client-id=...
+```
+
+3. End users identify their group memberships:
+
+```sh
+oc apply -o yaml -f - <<EOF
+apiVersion: authentication.k8s.io/v1
+kind: SelfSubjectReview
+metadata:
+  name: test
+EOF
+```
+
+Output:
+```sh
+apiVersion: authentication.k8s.io/v1
+kind: SelfSubjectReview
+metadata:
+  creationTimestamp: "2025-12-09T17:03:01Z"
+status:
+  userInfo:
+    ...
+    groups:
+    - system:authenticated
+    - foo # fetched from Microsoft Graph API endpoint
+```
+
+### API Extensions
+
+For review purposes, this section will focus on the user facing semantics of the API changes.
+The nuanced API review for the actual Go type changes will be done in a PR against the openshift/api
+repository (TODO).
+
+A new, optional, field `.spec.oidcProviders[].externalClaims` will be added to the
+`authentications.config.openshift` API.
+
+A YAML representation of the new API field and it's children:
+```yaml
+externalClaims:
+  # clientAuth is how the Kubernetes API server
+  # authenticates with the provided external claims sources.
+  # clientAuth is optional, and when not set will
+  # attempt to use anonymous authentication.
+  clientAuth:
```

Comment:
I realized that this is missing some sort of TLS configuration layer for configuring the trust store of the webhook authenticator when making requests to this external source.

I'll hold off on adding any of that until there is some sort of general consensus on whether or not this approach is a reasonable path to continue pursuing.